teps/0123-specifying-on-demand-retry-in-pipelinetask.md

```diff
@@ -0,0 +1,181 @@
+---
+status: proposed 
+title: Specifying on-demand-retry in a PipelineTask
+creation-date: '2022-09-16'
+last-updated: '2022-10-07'
+authors:
+- '@pritidesai'
+---
+
+# TEP-0123: Specifying on-demand-retry in a PipelineTask
+
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Requirements](#requirements)
+  - [Use Cases](#use-cases)
+    - [CD Use Case](#cd-use-case)
+    - [CI Use Case](#ci-use-case)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes a mechanism to allow users to choose when to retry a failed `pipelineTask` of a pipeline.
+This kind of `on-demand` retry of a `pipelineTask` is executed with the original `params` and `specifications`.
+The `on-demand` retry of a `pipelineTask` allow users to continue to retry until the task succeeds. After the offending task
+succeeds, the rest of the `pipeline` continues executing as usual until the completion.
```

Comment:
Thanks for the update @pritidesai - I think it would useful to specify from the start whether the retry you're talking about may take place after the completion of a `PipelineRun` or only during its original execution.

Reply:
thanks @afrittoli 👍 

I added an explanation for that saying its can only take place during its original execution.